jiant/tasks/qa.py

```diff
@@ -346,12 +349,12 @@ def _make_instance(psg, qst, ans_str, label, psg_idx, qst_idx, ans_idx):
             d["qst_idx"] = MetadataField(qst_idx)
```

Comment:
ReCoRD is the only task for which I'm still seeing a major performance regression in the refactored code, both with BERT and XLNet. @W4ngatang @pruksmhc - Would you mind giving this a look soon as a sanity check? I'm not really following the logic here.

For context, the likely source of the issue is that we're moving the insertion of [sep] and [cls] tokens to a new function, and that is happening here rather than earlier in preprocessing. BERT and XLNet place [sep] and [cls] in different positions.

Reply:
For ReCoRD, for each example we extract the possible given answer choices from the passage and construct "query-answer pairs" where we substitute each choice into the Cloze-style query. Then we concatenate the passage with the filled-in query to form an instance (so we're modeling each answer choice independently, mostly due to memory issues).

Reply:
ReCoRD seems resolved ?